src/RunAction.cpp

```diff
@@ -232,6 +233,11 @@ NodeBuildResult::Enum RunAction(BuildQueue *queue, ThreadState *thread_state, Ru
         {
             last_cmd_line = cmd_line;
             result = ExecuteProcess(cmd_line, env_count, env_vars, thread_state->m_Queue->m_Config.m_Heap, job_id, false, SlowCallback, &slowCallbackData);
+
+            if (silenceOnSuccess && result.m_ReturnCode == 0) {
```

Comment:
I think this is too early. We might have a tool which returns a 0 exit code, but that produces output that fails the AllowedOutputSubstrings check (in ValidateExecResultAgainstAllowedOuptut, below).

This said, are you really sure it's necessary to add a new flag for this? Most of our tools are silent without this flag existing. There's a codepath in ValidateExecResultAgainstAllowedOutput which returns `ValidationResult::SwallowStdout` - is it not possible to trigger that codepath without making changes to Tundra?

Reply:
Hmm, I would not have guessed that `AllowedOutputSubstrings` would swallow output on success- I think this deserves to be documented. (I assumed that if there was "unallowed" output then the command would be marked as failed, and then print the output.)

Now that I look, I think `ValidateExecResultAgainstAllowedOutput` might need some work- it doesn't appear to behave as documented in the unity tree (it will return `ValidationResult::SwallowStdout` if a there's a single match of any pattern, it does not explicitly check each line). But I need to take a closer look at the regex implementation to be certain.

Reply:
I think the original intent was to try and handle multiple use cases simultaneously:
1) This action should not produce any output. If it does, fail the action, because something weird is going on. (`allowUnexpectedOutput` is false, and there's no `allowedOutputSubstrings` provided).
2) This action produces some particular output messages that we don't care about, and if that's all it output, then we don't want to see them in stdout. However, if it produces any other output messages, then we want to fail the action because something weird is going on. (`allowUnexpectedOutput` is false, and one or more `allowedOutputSubstrings`  have been provided).
3) This command produces output, but we don't care what it is unless the action failed. (`allowUnexpectedOutput` doesn't matter, and `allowedOutputSubstrings` contains a regex that matches any output).

Case 3 is what we're facing with ILRepack I think. Looking at the XMLDoc for `IBackend.AddAction`, indeed the comments there do not match the implementation - saying that `allowedOutputSubstrings` will only be used when `allowUnexpectedOutput` is false is clearly not what's happening in the code.